lib/mrb/scripts/command_line/grndb.rb

```diff
@@ -99,8 +99,12 @@ def recover(database, options, arguments)
         begin
           recoverer.recover
         rescue Error => error
-          failed("Failed to recover database: <#{@database_path}>",
-                 error.message)
+          if options[:force_truncate]
```

Comment:
Database#recover raise an error so Recoverer#truncate_corrupt_objects will be skipped.
But as you pointed out, it seems that it is better to do in `recoverer.recover`.